index.bs

```diff
@@ -170,2193 +138,170 @@ The important commonality between them being that they offer some degree of spat
-};
-
-[SecureContext, Exposed=Window] interface XRRenderState {
-  readonly attribute double depthNear;
-  readonly attribute double depthFar;
-  readonly attribute double? inlineVerticalFieldOfView;
-  readonly attribute XRWebGLLayer? baseLayer;
-};
-</pre>
-
-Each {{XRRenderState}} has a <dfn for="XRRenderState">output canvas</dfn>, which is an {{HTMLCanvasElement}} initially set to <code>null</code>. The [=XRRenderState/output canvas=] is the DOM element that will display any content rendered for an {{XRSessionMode/"inline"}} {{XRSession}}.
-
-Each {{XRRenderState}} also has <dfn for="XRRenderState">composition disabled</dfn> boolean, which is initially <code>false</code>. The {{XRRenderState}} is considered to be have [=XRRenderState/composition disabled=] if rendering commands performed for an {{XRSessionMode/"inline"}} {{XRSession}} are executed in such a way that they are directly displayed into [=XRRenderState/output canvas=], rather than first being processed by the [=XR Compositor=].
-
-Note: At this point the {{XRRenderState}} will only have an [=XRRenderState/output canvas=] if it has [=XRRenderState/composition disabled=], but future versions of the specification are likely to introduce methods for setting [=XRRenderState/output canvas=]' that support more advanced uses like mirroring and layer compositing that will require composition.
-
-<div class="algorithm" data-algorithm="initialize-renderstate">
-
-When an {{XRRenderState}} object is created for an {{XRSession}} |session|, the user agent MUST <dfn>initialize the render state</dfn> by running the following steps:
-
-  1. Let |state| be the newly created {{XRRenderState}} object.
-  1. Initialize |state|'s {{XRRenderState/depthNear}} to <code>0.1</code>.
-  1. Initialize |state|'s {{XRRenderState/depthFar}} to <code>1000.0</code>.
-  1. Initialize |state|'s {{XRRenderState/inlineVerticalFieldOfView}} based on the following:
-    <dl class="switch">
-      <dt> If |session| is an [=immersive session=]
-      <dd> Initialize |state|'s {{XRRenderState/inlineVerticalFieldOfView}} to <code>null</code>.
-      <dt> Else
-      <dd> Initialize |state|'s {{XRRenderState/inlineVerticalFieldOfView}} to <code>PI * 0.5</code>.
-    </dl>
-  1. Initialize |state|'s {{XRRenderState/baseLayer}} to <code>null</code>.
-
-</div>
+Devices that lack one of the optional inputs listed in the tables above MUST preserve their place in the {{Gamepad/buttons}} or {{Gamepad/axes}} array, reporting a <dfn>placeholder button</dfn> or <dfn>placeholder axis</dfn>, respectively. A [=placeholder button=] MUST report <code>0</code> for {{GamepadButton/value}}, <code>false</code> for {{GamepadButton/pressed}}, and <code>false</code> for {{GamepadButton/touched}}. A [=placeholder axis=] MUST report <code>0</code>. [=Placeholder buttons=] and [=placeholder axis|axes=] MUST be omitted if they are the last element in the array or all following elements are also [=placeholder buttons=] or [=placeholder axis|axes=].
```

Comment:
Nit that looks like my error, but we might as well fix it here: Let's swap the first reference and definition of `placeholder button` and `placeholder axis` here so that the reference links actually points to the defining sentence, like so:

```
reporting a [=placeholder button=] or [=placeholder axis=], respectively. A <dfn>placeholder button</dfn> MUST report <code>0</code> for {{GamepadButton/value}}, <code>false</code> for {{GamepadButton/pressed}}, and <code>false</code> for {{GamepadButton/touched}}. A <dfn>placeholder axis</dfn> MUST report <code>0</code>.
```

Reply:
Not sure I follow why this needed to be changed?

Reply:
Not sure I follow why this needed to be changed?

Reply:
You know what, it's fine to leave as-is. I'll clean it up later.